packages/ra-auth-cognito/src/authProvider.ts

```diff
@@ -77,37 +93,83 @@ export const CognitoAuthProvider = (
               });
 
     return {
-        async login({
-            username,
-            password,
-            newPassword,
-            ...attributes
-        }: {
-            username: string;
-            password: string;
-            newPassword?: string;
-            [key: string]: any;
-        }) {
+        async login(form: FormData) {
             return new Promise((resolve, reject) => {
-                if (newPassword) {
+                const callback: IAuthenticationCallback = {
+                    onSuccess: result => {
+                        return resolve(result);
+                    },
+                    onFailure: err => {
+                        reject(err);
+                    },
+                    newPasswordRequired: () => {
+                        reject(new ErrorRequireNewPassword());
+                    },
+                    mfaSetup: () => {
+                        user.associateSoftwareToken({
+                            associateSecretCode: secretCode => {
+                                reject(
+                                    new ErrorMfaTotpAssociationRequired({
+                                        secretCode,
+                                        username: user.getUsername(),
+                                        applicationName:
+                                            config.applicationName ??
+                                            window.location.hostname,
+                                    })
+                                );
+                            },
+                            onFailure: err => {
+                                reject(err);
+                            },
+                        });
+                    },
+                    totpRequired: () => {
+                        reject(new ErrorMfaTotpRequired());
+                    },
+                    mfaRequired: () => {
+                        reject(new ErrorMFARequired());
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏

```suggestion
                        reject(new ErrorMFARequired('SMS MFA is required by the server, but it is not yet supported by ra-auth-cognito. Please disable this feature in Cognito config.'));
```